### PR TITLE
Simplify unit test results for pull requests by using GitHub CLI

### PR DESCRIPTION
**Pull request by jgiannuzzi**
Here is an example run for a PR from a fork: https://github.com/jgiannuzzi/publish-unit-test-result-action/runs/2667822522?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/unit-test-results.yml [hunk 1]
```diff
@@ -25,46 +25,21 @@ jobs:
...
           done
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
 
       - name: Publish Unit Test Results
         uses: EnricoMi/publish-unit-test-result-action@v1
         with:
           commit: ${{ github.event.workflow_run.head_sha }}
-          files: "artifacts/*/**/*.xml"
+          files: "artifacts/**/*.xml"
```
Comment (EnricoMi) [inline .github/workflows/unit-test-results.yml:45]: this hasn't been changed in README.md

File: README.md [hunk 2]
```diff
@@ -373,43 +373,18 @@ jobs:
          )
 
       steps:
-         - name: Download Artifacts
```
Comment (EnricoMi) [inline README.md:376]: keep in sync with review comments above

**Review by EnricoMi — CHANGES_REQUESTED** [review]
Welcome contributor! This is an awesome improvement! A few minor comments though.

File: .github/workflows/unit-test-results.yml [hunk 3]
```diff
@@ -25,46 +25,21 @@ jobs:
...
           do
-            if [ -f "$file" ]
-            then
-              dir="${file/%.zip/}"
-              mkdir -p "$dir"
-              unzip -d "$dir" "$file"
-            fi
+            name="$(jq -r .name <<<$artifact)"
+            gh api "$(jq -r .url <<<$artifact)" > "$name.zip"
+            unzip -d "$name" "$name.zip"
```
[the remark this answers is not in the document]
Comment (jgiannuzzi) [reply .github/workflows/unit-test-results.yml:36]: > does `unzip -d` create that directory?

yes, it does

> can we move `env:` before `run:`?

will do!

File: .github/workflows/unit-test-results.yml [hunk 4]
```diff
@@ -25,46 +25,21 @@ jobs:
...
-               fs.writeFileSync(artifact_path, Buffer.from(download.data));
-               console.log(`Downloaded ${artifact_path}`);
-            }
-      - name: Extract Artifacts
+      - name: Download and Extract Artifacts
         run: |
-          for file in artifacts/*.zip
+          mkdir artifacts && cd artifacts
+          IFS=$'\n'
+          for artifact in $(gh api ${{ github.event.workflow_run.artifacts_url }} -q '.artifacts[] | {name: .name, url: .archive_download_url}')
```
[the remark this answers is not in the document]
Comment (jgiannuzzi) [reply .github/workflows/unit-test-results.yml]: done

File: README.md (hunk 2 above)
Comment (jgiannuzzi) [reply README.md:376]: done

File: .github/workflows/unit-test-results.yml [hunk 5]
```diff
@@ -26,17 +26,22 @@ jobs:
...
       - name: Download and Extract Artifacts
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
         run: |
-          mkdir artifacts && cd artifacts
           IFS=$'\n'
-          for artifact in $(gh api ${{ github.event.workflow_run.artifacts_url }} -q '.artifacts[] | {name: .name, url: .archive_download_url}')
+
+          artifacts_url=${{ github.event.workflow_run.artifacts_url }}
+          artifacts=($(gh api $artifacts_url -q '.artifacts[] | {name: .name, url: .archive_download_url}'))
```
Comment (EnricoMi) [inline .github/workflows/unit-test-results.yml:35]: why are the outer `(` `)` needed here?

Comment (jgiannuzzi) [reply .github/workflows/unit-test-results.yml:35]: `artifacts` is built as a bash array

**Review by EnricoMi — APPROVED** [review]

File: .github/workflows/unit-test-results.yml (hunk 5 above)
Comment (jgiannuzzi) [reply .github/workflows/unit-test-results.yml]: done

**Comment by jgiannuzzi** [comment]
I noticed that the indentation style for the YAML snippets in the README was inconsistent, so I attempted to fix that in this PR too. Happy to open a different PR if you prefer.

**Review by EnricoMi — APPROVED** [review]
Awesome!

**Comment by EnricoMi** [comment]
@jgiannuzzi can you please create another PR from your fork to finally test these changes in my repo?

**Pull request merged**